File: apps/virtual-pet-adoption-server/src/APIs/Subscription/Dtos/Subscription.cs
using VirtualPetAdoption.Core.Enums;

namespace VirtualPetAdoption.APIs.Dtos;

public class Subscription
{
    public DateTime CreatedAt { get; set; }

    public DateTime? EndDate { get; set; }

    public string Id { get; set; }

    public string? Pet { get; set; }

    public DateTime? StartDate { get; set; }

    public StatusEnum? Status { get; set; }

    public DateTime UpdatedAt { get; set; }

    public string? User { get; set; }
}

File: apps/virtual-pet-adoption-js-admin/src/api/subscription/Subscription.ts
import { Pet } from "../pet/Pet";
import { User } from "../user/User";

export type Subscription = {
  createdAt: Date;
  endDate: Date | null;
  id: string;
  pet?: Pet | null;
  startDate: Date | null;
  status?: "Option1" | null;
  updatedAt: Date;
  user?: User | null;
};
